aspnetcore/fundamentals/dependency-injection.md

```diff
@@ -146,54 +209,160 @@ Transient lifetime services are created each time they're requested. This lifeti
 Scoped lifetime services are created once per request.
 
 > [!WARNING]
-> If you're using a scoped service in a middleware, inject the service into the `Invoke` or `InvokeAsync` method. Don't inject via constructor injection because it forces the service to behave like a singleton.
+> When using a scoped service in a middleware, inject the service into the `Invoke` or `InvokeAsync` method. Don't inject via constructor injection because it forces the service to behave like a singleton. For more information, see <xref:fundamentals/middleware/index>.
 
 **Singleton**
 
-Singleton lifetime services are created the first time they're requested (or when `ConfigureServices` is run if you specify an instance there) and then every subsequent request will use the same instance. If your application requires singleton behavior, allowing the services container to manage the service's lifetime is recommended instead of implementing the singleton design pattern and managing your object's lifetime in the class yourself.
+Singleton lifetime services are created the first time they're requested (or when `ConfigureServices` is run and an instance is specified with the service registration). Every subsequent request uses the same instance. If the app requires singleton behavior, allowing the service container to manage the service's lifetime is recommended. Don't implement the singleton design pattern and provide user code to manage the object's lifetime in the class.
+
+> [!WARNING]
+> It's dangerous to resolve a scoped service from a singleton. It may cause the service to have incorrect state when processing subsequent requests.
+
+### Constructor injection behavior
+
+Services can be resolved by two mechanisms:
+
+* `IServiceProvider`
+* [ActivatorUtilities](/dotnet/api/microsoft.extensions.dependencyinjection.activatorutilities)
+
+Constructors can accept arguments that aren't provided by dependency injection, but the arguments must assign default values.
+
+When services are resolved by `IServiceProvider` or `ActivatorUtilities`, constructor injection requires a *public* constructor.
+
+When services are resolved by `ActivatorUtilities`, constructor injection requires that only one applicable constructor exist. Constructor overloads are supported, but only one overload can exist whose arguments can all be fulfilled by dependency injection.
```

Comment:
Needs more info where and why ActivatorUtilities is used

Reply:
@pakrym I see the API remarks and a few code comments here and there (e.g., Hosting repo for `StartupLoader.LoadMethods`), which don't seem quite adequate to just drop in here. I'd prefer if you or someone from engineering write a paragraph for this or even just supply a handful of sentences that I can wordsmith into a paragraph.

There's probably a need for an advanced DI topic that goes beyond the basics laid out in this topic. I'm aware of `@stevejgordon`'s [Expand DI documentation with additional extension method overloads (aspnet/Docs \#6312)](https://github.com/aspnet/Docs/issues/6312), but it feels to me like we need something between this topic and those. I propose we open a new issue for it tho ... we're swamped at the moment.